### ISSUE TITLE
After removing TSParenthesizedType, there was no way to detect the presence of parentheses.

### ISSUE THREAD

**Issue by foul11** [typescript-eslint/typescript-eslint#12149]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&showAST=es&fileType=.tsx&code=C4TwDgpgBAglC8UDkBDJBuAsAKFJKAQgsgEYY457QDCxOUUAZLPU1ANoEC6UEAHsAgA7ACYBnDkIgA3CACcurBgH4owOQFcISqAC4oAMxQAbMduwMGlcNAAidC1AAUzGK2acuASl4Dh4yRl5RUcVNU1zSz1DEzMda3wAMQcGV3dnT19BUQl2KVkFHVV1LR19I1MILyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type A = 'a';
type B = 'b';

type C = 
  & A
  & [B] extends [never]
    ? true
    : false
    
type D = 
  (& A
  & [B]) extends [never]
    ? true
    : false
    
type F = 
  & A
  & ([B] extends [never]
    ? true
    : false)
```

### Expected Result

I expected to be able to distinguish type C from type D.

### Actual Result

I couldn't do this because there isn't enough information in the AST about the presence of brackets.

### Additional Info

I wanted to create a custom rule that prohibits placing TSConditionalType without parentheses next to TSIntersectionType and TSUnionType.

```
"no-restricted-syntax": [ "error", [
  "TSConditionalType[checkType.type=\"TSIntersectionType\"]",
]],
```

I found an explanation for this in issue #3136, which asked to remove the TSParenthesizedType node.
After this, it's impossible to check for parentheses or distinguish between type D and type C.
Perhaps it would be worth adding some flags that would indicate that the type was additionally enclosed in brackets?

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.34.0` |
| `@typescript-eslint/parser`            | `8.34.0` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `22.20.0` |

**Comment by bradzacher** [comment]
This is intentional.
Parentheses are never encoded into the AST.

Similarly if you have `const x = (1+1)` the parentheses are not encoded into the AST.

We will not be adding flags to denote the presence of parentheses.

If you want to detect the presence of the you can check for them by creating a proper lint rule and checking the tokens around the nodes.
